manifests/vhost.pp

```diff
@@ -251,8 +251,12 @@
   # This ensures that the docroot exists
   # But enables it to be specified across multiple vhost resources
   if ! defined(File[$docroot]) and $manage_docroot {
+    $docroot_ensure = $ensure? {
+      present => 'directory',
+      default => 'absent',
+    }
```

Comment:
The issue with defaulting to have the `$docroot_ensure` value default to `$ensure` is that the module supports multiple vhosts in the same docroot, so you could end up accidentally blowing away other things. I think I like this being added as a parameter with a default of `directory`, and then the user just needs to be explicit if they want to have the docroot removed.